extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/drive/DriveExporter.java

```diff
@@ -0,0 +1,204 @@
+import org.datatransferproject.spi.transfer.types.ContinuationData;
+import org.datatransferproject.spi.transfer.types.ExportInformation;
+import org.datatransferproject.spi.transfer.types.IdOnlyContainerResource;
+import org.datatransferproject.spi.transfer.types.StringPaginationToken;
+import org.datatransferproject.types.transfer.auth.TokensAndUrlAuthData;
+import org.datatransferproject.types.transfer.models.blob.BlobbyStorageContainerResource;
+import org.datatransferproject.types.transfer.models.blob.DigitalDocument;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DriveExporter implements
+    Exporter<TokensAndUrlAuthData, BlobbyStorageContainerResource> {
+  private static final Logger logger = LoggerFactory.getLogger(DriveExporter.class);
+  private static final String DRIVE_QUERY_FORMAT  = "'%s' in parents and trashed=false";
+  static final String FOLDER_MIME_TYPE = "application/vnd.google-apps.folder";
+  private static final String FUSION_TABLE_MIME_TYPE = "application/vnd.google-apps.fusiontable";
+  private static final String MAP_MIME_TYPE = "application/vnd.google-apps.map";
+  // This is generated from calling https://www.googleapis.com/drive/v3/about and
+  // picking the best default.
+  private static final ImmutableMap<String, String> EXPORT_FORMATS =
+      ImmutableMap.<String, String>builder()
+          .put(
+              "application/vnd.google-apps.document",
+              "application/vnd.openxmlformats-officedocument.wordprocessingml.document")
+          .put(
+              "application/vnd.google-apps.spreadsheet",
+              "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet")
+          .put(
+              "application/vnd.google-apps.drawing",
+              "image/svg+xml")
+          .put(
+              "application/vnd.google-apps.presentation",
+              "application/vnd.openxmlformats-officedocument.presentationml.presentation")
+          .put(
+              "application/vnd.google-apps.script",
+              "application/vnd.google-apps.script+json")
+          .put(
+              "application/vnd.google-apps.jam",
+              "application/pdf"
+          )
+          .put(
+              "application/vnd.google-apps.form",
+              "application/zip"
+          )
+          .put(
+              "application/vnd.google-apps.site",
+              "text/plain"
+          )
+          .build();
+
+  private final GoogleCredentialFactory credentialFactory;
+  private volatile Drive driveInterface;
+  private final JobStore jobStore;
+
+  public DriveExporter(GoogleCredentialFactory credentialFactory, JobStore jobStore) {
+    this(credentialFactory,
+        // Lazily initialized later on
+        null,
+        jobStore);
+  }
+
+  @VisibleForTesting
+  DriveExporter(
+      GoogleCredentialFactory credentialFactory,
+      Drive driveInterface,
+      JobStore jobStore) {
+    this.credentialFactory = credentialFactory;
```

Comment:
should we checkNotNull on the credentialFactory too? and mark driveInterface @Nullable?